src/blame.rs

```diff
@@ -0,0 +1,512 @@
+      git_opts.use_mailmap(use_mailmap);
+    }
+
+    git_opts
+  }
+}
+
+#[napi]
+pub struct Blame {
+  pub(crate) inner: BlameInner,
+}
+
+pub(crate) enum BlameInner {
+  Repo(SharedReference<Repository, git2::Blame<'static>>),
+}
+
+impl Deref for BlameInner {
+  type Target = git2::Blame<'static>;
+
+  fn deref(&self) -> &Self::Target {
+    match self {
+      Self::Repo(repo) => repo.deref(),
+    }
+  }
+}
+
+/// Iterator over blame hunks.
+pub struct BlameIter<'a> {
+  blame: &'a Blame,
+  idx: usize,
+  len: usize,
+}
+
+impl Iterator for BlameIter<'_> {
+  type Item = Result<BlameHunk>;
+
+  fn next(&mut self) -> Option<Self::Item> {
+    if self.idx >= self.len {
+      return None;
+    }
+
+    let result = self.blame.get_hunk_by_index(self.idx as u32);
+    self.idx += 1;
+
+    Some(result)
+  }
+
+  fn size_hint(&self) -> (usize, Option<usize>) {
+    let remaining = self.len - self.idx;
+    (remaining, Some(remaining))
+  }
+}
+
+impl ExactSizeIterator for BlameIter<'_> {
+  fn len(&self) -> usize {
+    self.len - self.idx
+  }
+}
+
+#[napi]
+impl Blame {
+  #[napi]
+  /// Gets the number of hunks in the blame result
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   getHunkCount(): number;
+  /// }
+  /// ```
+  ///
+  /// @returns The number of hunks in the blame result
+  pub fn get_hunk_count(&self) -> u32 {
+    self.inner.len() as u32
+  }
+
+  #[napi]
+  /// Checks if the blame result is empty
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   isEmpty(): boolean;
+  /// }
+  /// ```
+  ///
+  /// @returns True if the blame result contains no hunks
+  pub fn is_empty(&self) -> bool {
+    self.inner.is_empty()
+  }
+
+  /// Returns an iterator over the hunks in this blame.
+  ///
+  /// This internal method is used to implement getHunks() and is not
+  /// directly exposed to JavaScript.
+  pub fn iter(&self) -> BlameIter {
+    BlameIter {
+      blame: self,
+      idx: 0,
+      len: self.get_hunk_count() as usize,
+    }
+  }
+
+  #[napi]
+  /// Generates blame information from an in-memory buffer
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   buffer(buffer: Buffer, bufferLen: number): Blame;
+  /// }
+  /// ```
+  ///
+  /// @example
+  /// ```ts
+  /// const buffer = Buffer.from('modified content');
+  /// const bufferBlame = blame.buffer(buffer, buffer.length);
+  /// ```
+  ///
+  /// @param {Buffer} buffer - Buffer containing file content to blame
+  /// @param {number} buffer_len - Length of the buffer in bytes
+  /// @returns A new Blame object for the buffer content
+  /// @throws If the buffer contains invalid UTF-8
+  pub fn buffer(&self, buffer: Buffer, buffer_len: u32, env: Env) -> Result<Blame> {
+    let content = std::str::from_utf8(&buffer[..buffer_len as usize])
+      .map_err(|e| Error::from_reason(format!("Invalid UTF-8 in buffer: {}", e)))?;
```

Comment:
```suggestion
  pub fn buffer(&self, buffer: Buffer, buffer_len: u32, env: Env) -> crate::Result<Blame> {
    let content = std::str::from_utf8(&buffer[..buffer_len as usize])?;
```

Can we use `crate::Result` for all methods?
We have implemented the frequently used errors so that we can eliminate the use of `.map_err`.